catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrFilterDelegate.java

```diff
@@ -300,22 +300,34 @@ public SolrQuery propertyIsLike(String propertyName, String pattern, boolean isC
 
     String searchPhrase = escapeSpecialCharacters(pattern);
 
-    if (Metacard.ANY_TEXT.equals(propertyName) && SOLR_WILDCARD_CHAR.equals(searchPhrase)) {
+    boolean isAnyText = Metacard.ANY_TEXT.equals(propertyName);
+
+    if (isAnyText && SOLR_WILDCARD_CHAR.equals(searchPhrase)) {
       return new SolrQuery("*:*");
     }
 
-    if (searchPhrase.contains(SOLR_WILDCARD_CHAR)
-        || searchPhrase.contains(SOLR_SINGLE_WILDCARD_CHAR)) {
+    boolean isWildcard =
+        searchPhrase.contains(SOLR_WILDCARD_CHAR)
+            || searchPhrase.contains(SOLR_SINGLE_WILDCARD_CHAR);
+
+    boolean isQuotedPhrase = pattern.startsWith("\"") && pattern.endsWith("\"");
+
+    if (isQuotedPhrase) {
+      searchPhrase =
+          searchPhrase.substring(2, searchPhrase.length() - 2).replaceAll("\\s", "\\\\ ");
+      if (!isWildcard) {
+        searchPhrase = SOLR_WILDCARD_CHAR + searchPhrase + SOLR_WILDCARD_CHAR;
+      }
+    } else if (isWildcard) {
       searchPhrase = "(" + searchPhrase + ")";
     } else {
       // Not an exact phrase
       searchPhrase = QUOTE + searchPhrase + QUOTE;
     }
 
-    if (searchPhrase.contains(SOLR_WILDCARD_CHAR)
-        || searchPhrase.contains(SOLR_SINGLE_WILDCARD_CHAR)
-        || Metacard.ANY_TEXT.equals(propertyName)) {
-
+    if (isQuotedPhrase) {
+      return new SolrQuery(wildcardSolrQuery(searchPhrase, propertyName, isCaseSensitive, true));
+    } else if (isWildcard || isAnyText) {
       return new SolrQuery(wildcardSolrQuery(searchPhrase, propertyName, isCaseSensitive, false));
```

Comment:
❓ Could the query returns for `isQuotedPhrase` and `isWildcard` now get moved to the earlier `if` block? Actually I think the logic could be simplified as separate blocks now: 
```
if (isQuotedPhrase) {
... 
return new SolrQuery(...);
}

if (isWildcard) {
...
return new SolrQuery(...);
}

// Not an exact phrase
searchPhrase = QUOTE + searchPhrase + QUOTE;

if (isCaseSensitive) {
...
}
return new SolrQuery(...);
```

I realize this might be a matter of preference, but the decreased nesting would help me consume this faster. Assuming I'm correct this is a valid refactor and I'm not missing something.

Reply:
I did try to do that but it did not work.  The exact phrase case is needed for anyText cases which have the same behavior as wildcard though wildcard cases are escaped differently.

Reply:
I did unnested all the other cases I could.